src/OptimalFilter.jl

```diff
@@ -45,15 +46,13 @@ struct Grid{T}
     y_length::T
 end
 
-# Covariance function r(x,y), equation 1 in Dietrich and Newsam 96
-function covariance(x::T, y::T, noise_params::NamedTuple) where T
-
-    return noise_params.sigma^2 * exp(-(abs(x) + abs(y))/(2 * noise_params.lambda))
-
+# Covariance function r(x,y), equation 1 in Dietrich & Newsam 1996
+function covariance(x::T, y::T, covariance_structure::IsotropicCovarianceStructure{T}) where T
+    return apply(covariance_structure, [x, y])
```

Comment:
What's `apply`? :thinking:

Reply:
Yes sorry should have made that more explicit.